src/Module/User/Import.php

```diff
@@ -0,0 +1,415 @@
+						$contact['archive'] = '1';
+				}
+			}
+
+			$contact['uid'] = $newUid;
+			if ($this->dbImportAssoc('contact', $contact) === false) {
+				$this->logger->warning('Error inserting contact', ['nick' => $contact['nick'], 'network' => $contact['network'], 'error' => $this->database->errorMessage()]);
+				$errorCount++;
+			} else {
+				$contact['newid'] = $this->lastInsertId();
+			}
+		});
+
+		if ($errorCount > 0) {
+			$this->systemMessages->addNotice($this->tt('%d contact not imported', '%d contacts not imported', $errorCount));
+		}
+
+		array_walk($account['group'], function (&$group) use ($newUid) {
+			$group['uid'] = $newUid;
+			if ($this->dbImportAssoc('group', $group) === false) {
+				$this->logger->warning('Error inserting group', ['name' => $group['name'], 'error' => $this->database->errorMessage()]);
+			} else {
+				$group['newid'] = $this->lastInsertId();
+			}
+		});
+
+		foreach ($account['group_member'] as $group_member) {
+			$import = 0;
+			foreach ($account['group'] as $group) {
+				if ($group['id'] == $group_member['gid'] && isset($group['newid'])) {
+					$group_member['gid'] = $group['newid'];
+					$import++;
+					break;
+				}
+			}
+
+			foreach ($account['contact'] as $contact) {
+				if ($contact['id'] == $group_member['contact-id'] && isset($contact['newid'])) {
+					$group_member['contact-id'] = $contact['newid'];
+					$import++;
+					break;
+				}
+			}
+
+			if ($import == 2 && $this->dbImportAssoc('group_member', $group_member) === false) {
+				$this->logger->warning('Error inserting group member', ['gid' => $group_member['id'], 'error' => $this->database->errorMessage()]);
+			}
+		}
+
+		foreach ($account['profile'] as $profile) {
+			unset($profile['id']);
+			$profile['uid'] = $newUid;
+
+			array_walk($profile, function (&$field) use ($oldUid, $oldBaseUrl, $oldAddr, $newUid, $newBaseUrl, $newAddr) {
+				$field = str_replace([$oldBaseUrl, $oldAddr], [$newBaseUrl, $newAddr], $field);
+				foreach (['profile', 'avatar'] as $key) {
+					$field = str_replace($oldBaseUrl . '/photo/' . $key . '/' . $oldUid . '.jpg', $newBaseUrl . '/photo/' . $key . '/' . $newUid . '.jpg', $field);
+				}
+			});
+
+			if (count($account['profile']) === 1 || $profile['is-default']) {
+				if ($this->dbImportAssoc('profile', $profile) === false) {
+					$this->logger->warning('Error inserting profile', ['error' => $this->database->errorMessage()]);
+					$this->systemMessages->addNotice($this->t('User profile creation error'));
+					$this->database->delete('user', ['uid' => $newUid]);
+					$this->database->delete('profile_field', ['uid' => $newUid]);
+					return;
+				}
+
+				$profile['id'] = $this->database->lastInsertId();
+			}
+
+			Profile::migrate($profile);
+		}
+
+		$permissionSet = $this->permissionSet->selectDefaultForUser($newUid);
+
+		foreach ($account['profile_fields'] ?? [] as $profile_field) {
+			$profile_field['uid'] = $newUid;
+
+			///@TODO Replace with permissionset import
```

Comment:
Do you want to address this todo? Or don't we need it yet?

Reply:
We need to export the permissions first, which isn't in the scope of this move just yet.